### PR TITLE
fixed the gaussian mixture example

### PR DESCRIPTION
**Pull request by h3jia**
Not sure if this is pyro/torch-version dependent, but I found that if I gave a bunch of theta's at the same time, the samples will either all from the smaller Gaussian or all from the larger Gaussian.

I think the issue is that it uses the same single call of `pdist.Categorical` regardless of the input theta size. With a simple fix, now the samples look correct on my side.

### CODE REVIEW

**Comment by janfb** [comment]
Thanks for reporting this and suggesting a fix @h3jia 

I can confirm that the current implementation selects only a single index independently of the number of passed parameters. 

@jan-matthis I suggest to label this as a bug. Can you request me as a reviewer then I can review the fix suggested by @h3jia 

In agree with the fix, I would only frame it slightly different and add documentation: 
```python
def simulator(parameters):
            # Sample mixture index for each parameter in batch
            idx = pyro.sample(
                "mixture_idx",
                pdist.Categorical(
                    probs=self.simulator_params["mixture_weights"].repeat(
                        parameters.shape[0], 1
                    )
                ),
            ).unsqueeze(1)

            # Select loc and scales according to mixture index
            scale = self.simulator_params["mixture_scales"][idx]
            loc = self.simulator_params["mixture_locs_factor"][idx] * parameters

            return pyro.sample("data", pdist.Normal(loc=loc, scale=scale))

```

File: sbibm/tasks/gaussian_mixture/task.py [hunk 1]
```diff
@@ -69,15 +69,17 @@ def get_simulator(self, max_calls: Optional[int] = None) -> Simulator:
...
+                pdist.Categorical(probs=torch.tile(self.simulator_params["mixture_weights"][None],
+                                  (parameters.shape[0], 1)))
             )
             return pyro.sample(
                 "data",
                 pdist.Normal(
-                    loc=self.simulator_params["mixture_locs_factor"][idx] * parameters,
-                    scale=self.simulator_params["mixture_scales"][idx],
+                    loc=self.simulator_params["mixture_locs_factor"][idx][:, None] * parameters,
+                    scale=self.simulator_params["mixture_scales"][idx][:, None],
```
Comment (janfb) [inline sbibm/tasks/gaussian_mixture/task.py:82]: I think here in `line 83` we would need a `.to_event(1)` call on the `Normal` to have the correct batch and event dimensions.

**Review by janfb — COMMENTED** [review]
Thanks a lot for reporting and fixing this! 
There is one change required to make this work, see comment below.

**Comment by jan-matthis** [comment]
@h3jia thank you very much for reporting this! Merged #63 in favour of this PR, which adds a test, and [released the fix as part of `v1.1.0`](https://github.com/sbi-benchmark/sbibm/releases/tag/v1.1.0).

Since experiments in the paper were run with a simulation batch size of 1000, this has an effect on results. We will issue an update to account for this.

**Comment by h3jia** [comment]
@jan-matthis Glad to hear that this has been merged. Can I ask some related questions: what's the optimizer setup for the benchmark results (or where can I find the info if it's somewhere)? Did you tune e.g. the learning rates for each individual problem and method, or do you just use some universal value?

And you mentioned that the batch size was 1000; is this only for the 100k budget cases? I guess you'd use a smaller batch size if you only have a budget of 1k-10k.

**Comment by jan-matthis** [comment]
> @jan-matthis Glad to hear that this has been merged. Can I ask some related questions: what's the optimizer setup for the benchmark results (or where can I find the info if it's somewhere)? Did you tune e.g. the learning rates for each individual problem and method, or do you just use some universal value?

Sure! All configs are in https://github.com/sbi-benchmark/results/tree/main/benchmarking_sbi/config. We did not tune hyperparameters per task but rather across tasks. Our rationale is in the paper, section 4, last paragraph.

> And you mentioned that the batch size was 1000; is this only for the 100k budget cases? I guess you'd use a smaller batch size if you only have a budget of 1k-10k.

`simulation_batch_size` generally was 1000, but note that it can get clipped. For example, when running a neural sequential algorithm, it gets clipped to maximally be `num_simulations_per_round`.
